src/metabase/models/params/chain_filter.clj

```diff
@@ -517,20 +524,22 @@
     field-id
     (field-values/field-should-have-field-values? field-id)))
 
-(defn- cached-field-values [field-id constraints {:keys [limit]}]
+(defn- cached-field-values [field constraints {:keys [limit]}]
   ;; TODO: why don't we remap the human readable values here?
   (let [{:keys [values has_more_values]}
         (if (empty? constraints)
-          (params.field-values/get-or-create-field-values-for-current-user! (t2/select-one Field :id field-id))
-          (params.field-values/get-or-create-linked-filter-field-values! (t2/select-one Field :id field-id) constraints))]
+          (params.field-values/get-or-create-field-values-for-current-user! (t2/select-one Field :id (:field-id field)))
+          (params.field-values/get-or-create-linked-filter-field-values! (t2/select-one Field :id (:field-id field)) constraints
+                                                                         (fn []
+                                                                           (unremapped-chain-filter field constraints {}))))]
```

Comment:
This allows to remove circular dependency plus instead of passing `field :- Constraint` into `params.field-values/*`, just keep it to `params.chain-filter` ns.

But that breaks some `enterprise.sandbox` calls, and I haven't solved that yet.

Reply:
is that necessary for this PR? I'd leave that to a separate PR if we can

Reply:
It is necessary, since in here
```
((resolve 'metabase.models.params.chain-filter/unremapped-chain-filter) (:id field) constraints {})
 ```
 
`field` is not a Constraint, it's a field from db, so it has no information about joins and when caching is enabled changes stop working. And passing a function feels no worse than passing additional information about field (and polluting `field-values` with it).